sdk/storage/Azure.Storage.Blobs/tests/BlobClientTests.cs

```diff
@@ -742,6 +743,30 @@ public async Task UploadAsync_OverwritesDeliberately_Path()
                 }
             }
         }
+
+        [LiveOnly]
+        [Test]
+        public async Task UploadAsync_ProgressReporting()
+        {
+            // Arrange
+            await using DisposingContainer test = await GetTestContainerAsync();
+            BlobClient blob = InstrumentClient(test.Container.GetBlobClient(GetNewBlobName()));
+            Progress progress = new Progress();
+            using var stream = new MemoryStream(GetRandomBuffer(Constants.GB));
+
+            // Act
+            await blob.UploadAsync(content: stream, progressHandler: progress);
```

Comment:
Can we change the `StorageTransferOptions` to do many workers with small transfer blocks?  The only thing I worry about with this change is the thread safety of the progress reporting code paths.

Reply:
Done.  The tests still pass.